build failure on clang-14, 15, 16

DOSBox-X 2024.03.01 uses ESFMu v1.2.3 and I received a report about build failure on OpenBSD-current/i386.

I tested building ESFMu with clang-13 to 16, only clang-13 can build ESFMu (Slackware-15.0/i686 have clang-13).

Here is 32bit code build test result on Debian-12/amd64.
```
uaa@emeraude:~/ESFMu$ clang-13 -m32 esfm.c -c -o /dev/null
uaa@emeraude:~/ESFMu$ clang-14 -m32 esfm.c -c -o /dev/null
esfm.c:1849:5: error: inline assembly requires more registers than available
                                "movzbl  %b[wave], %%eax             \n\t"
                                ^
esfm.c:1849:5: error: inline assembly requires more registers than available
2 errors generated.
uaa@emeraude:~/ESFMu$ clang-15 -m32 esfm.c -c -o /dev/null
esfm.c:1849:5: error: inline assembly requires more registers than available
                                "movzbl  %b[wave], %%eax             \n\t"
                                ^
esfm.c:1849:5: error: inline assembly requires more registers than available
2 errors generated.
uaa@emeraude:~/ESFMu$ clang-16 -m32 esfm.c -c -o /dev/null
esfm.c:1849:5: error: inline assembly requires more registers than available
                                "movzbl  %b[wave], %%eax             \n\t"
                                ^
esfm.c:1849:5: error: inline assembly requires more registers than available
2 errors generated.
uaa@emeraude:~/ESFMu$
```
I have no idea to fix this, so simply report this as issue.

--- a/esfm.c
+++ b/esfm.c
@@ -1845,10 +1845,13 @@ ESFM_process_feedback(esfm_chip *chip)
 				: "cc", "ax", "bx", "cx", "dx", "r8", "r9", "r10", "r11"
 			);
 #elif defined(__GNUC__) && defined(__i386__) && !defined(_ESFMU_DISABLE_ASM_OPTIMIZATIONS)
+			size_t logsinrom_addr = (size_t)logsinrom;
+			size_t exprom_addr = (size_t)exprom;
+
 			asm (
 				"movzbl  %b[wave], %%eax             \n\t"
 				"shll    $11, %%eax                  \n\t"
-				"leal    %[sinrom], %%edi            \n\t"
+				"movl    %[sinrom], %%edi            \n\t"
 				"addl    %%eax, %%edi                \n\t"
 				"shlw    $3, %[eg_out]               \n\t"
 				"xorl    %[out], %[out]              \n\t"
@@ -1883,7 +1886,7 @@ ESFM_process_feedback(esfm_chip *chip)
 				// wave_out = exprom[level & 0xff] >> (level >> 8);
 				"movb    %%ah, %%cl                  \n\t"
 				"movzbl  %%al, %%eax                 \n\t"
-				"leal    %[exprom], %[out]           \n\t"
+				"movl    %[exprom], %[out]           \n\t"
 				"movzwl  (%[out], %%eax, 2), %[out]  \n\t"
 				"shrl    %%cl, %[out]                \n\t"
 				// if (lookup & 0x8000) wave_out = -wave_out;
@@ -1905,8 +1908,8 @@ ESFM_process_feedback(esfm_chip *chip)
 				: [p_off]  "m"   (phase_offset),
 				  [mod_in] "m"   (mod_in_shift),
 				  [wave]   "m"   (waveform),
-				  [sinrom] "m"   (logsinrom),
-				  [exprom] "m"   (exprom),
+				  [sinrom] "m"   (logsinrom_addr),
+				  [exprom] "m"   (exprom_addr),
 				  [i]      "m"   (iter_counter)
 				: "cc", "ax", "bx", "cx", "di"
 			);

